packages/ui-causal-graph-editor/src/graph-viewer/causal-graph-editor.tsx

```diff
@@ -158,6 +158,22 @@ function CausalGraphEditor(props: CausalGraphEditorProps): JSX.Element {
         setError(e);
     };
 
+    const isTimeSeriesCausalGraph = useMemo(() => {
+        const nodeClass = state.graph.getNodeAttribute(state.graph.nodes()[0], 'extras')?.node_class;
+        return nodeClass === 'TimeSeriesNode';
+    }, [state.graph]);
+
+    // If we have a time series causasl graph and if the layout chosen supports tiers and these are not defined
+    if (
+        isTimeSeriesCausalGraph &&
+        'tiers' in props.graphLayout &&
+        'orientation' in props.graphLayout &&
+        props.graphLayout.tiers === undefined
+    ) {
+        // We set tiers based on TimeSeriesCausalGraph structure of node lags
+        props.graphLayout.tiers = { group: 'variable_name', order_nodes_by: 'time_lag' };
+    }
```

Comment:
I'm not sure if this is the best place for this logic to live. Should probably be moved within `useCausalGraphEditor` not to pollute the main component. Should also only run once on each graph instance, ideally, whereas here it's running on each render.

Reply:
Makes sense, have moved the logic to useCausalGraphEditor and added within a useEffect so it will only run if it stops being a timeseries or the layout changes